db/service/src/main/java/org/commonjava/indy/db/service/ServiceEventHandler.java

```diff
@@ -120,4 +114,30 @@ else if ( store.getType() == StoreType.group )
         } );
 
     }
+
+    private void updateCaches( final ArtifactStore newStore, final StoreKey storeKey,
+                               final BasicCacheHandle<StoreKey, ArtifactStore> storeCache,
+                               final BasicCacheHandle<Object, Collection<ArtifactStore>> queryCache )
+    {
+        if ( storeCache.get( storeKey ) != null )
+        {
+            storeCache.remove( storeKey );
+        }
+        storeCache.put( newStore.getKey(), newStore );
+
+        queryCache.execute( ( cache ) -> {
+            for ( Collection<ArtifactStore> stores : cache.values() )
+            {
+                for ( ArtifactStore store : stores )
+                {
+                    if ( store.getKey().equals( storeKey ) )
+                    {
+                        stores.remove( store );
+                    }
+                    stores.add( newStore );
```

Comment:
This is not correct logic. For queryCache, it contains stores for artifact query (like affected by). If a cache query result does not contain the new added store, this logic will cause a incorrect query result. I think this "stores.add(newStore)" should also be put in to the "if(store.getKey().equals(storeKey)" after the removal.

Reply:
@ligangty, Ok, I push another commit: https://github.com/Commonjava/indy/pull/2168/commits/2795795f5352f552b51a6971bfb24402817d795d, could you help to review whether it's proper for query cache logic?

Reply:
@yma88 I'm still confused. Why it is still needed to add here? The only case to update the cache is the key is matched, or no need to do anything.

Reply:
@ligangty, this might be related to the case I tested: I found in the group maven-metadata.xml, hosted version failed to remove after hosted repo disabled, and in log, the hosted repo's affected group is null, which should be caused by the query cache is not up to date, https://github.com/Commonjava/indy/blob/master/core/src/main/java/org/commonjava/indy/core/change/StoreContentListener.java#L260, so for postUpdate event, beside the UPDATE type, we also need to add the new stores in cache for ADD type, that's what I caught.